### ISSUE TITLE
React parent components stay unchanged when using I18n.changeLanguage(lng)

### ISSUE THREAD

**Issue by wheelo** [i18next/react-i18next#611]
First of all, the design of your plugin system is delicate.

While in my React application, I use **I18n.changeLanguage(lng)** which works great at updating and re-rendering any **children** components those bind I18n.t('key'). But currently all the parent components stay unchanged when the **I18n.changeLanguage** triggered in children components.

**To Reproduce**
Change the selected Language to **Francais** at the bottom, only the language at the bottom area switches to other language, while elsewhere stays unchanged.

**Expected behaviour**
Every component would be changed from top to bottom when using the I18n.changeLanguage(lng). Is that the problem involved with some global event emitting system?

**Screenshots**
<img width="595" alt="default" src="https://user-images.githubusercontent.com/3896369/48173947-d1ad2e80-e340-11e8-88ec-796c87020000.png">

**Context**
All the React components in my application have been decorated with the **withT** high order function, and the t('key') method has been used to translate the inner text .

**Comment by isaachinman** [comment]
Hi @wheelo. You're going to need to explain your implementation if you want help. Are you using `NamespacesConsumer`, or `I18nextProvider`, or something else?

If you believe you have encountered a genuine bug, and it's as simple as you say, it should be quite easy for you to create and share a reproducible repo. Thanks.

**Comment by wheelo** [comment]
I18nextProvider is used in the entry

**Comment by isaachinman** [comment]
What entry? I see you gave steps to reproduce, and uploaded a GIF, but you didn't link to any codebase. We have nothing to go on so far.

**Comment by wheelo** [comment]
@isaachinman Thanks, I am trying to make a simple demo to reproduce this problem

**Comment by isaachinman** [comment]
Okay, got it. Let me know when you do, and I'll take a look.

**Comment by wheelo** [comment]
Lunch time in China

**Comment by isaachinman** [comment]
Take a look at the [docs here](https://react.i18next.com/components/withi18n):

<img width="774" alt="screenshot 2018-11-07 at 19 32 12" src="https://user-images.githubusercontent.com/10575782/48176182-daabea80-e2c3-11e8-9c92-e533e0d073a9.png">

Let me know if you have any other questions.

**Comment by isaachinman** [comment]
@wheelo I'm not seeing any of the content being translated in the first place.

**Comment by isaachinman** [comment]
You weren't exporting your resources correctly from `resources.js`.

**Comment by isaachinman** [comment]
So:

1. You don't need to, nor should you, wrap your entire `App` into this `AppTrans`. The `withI18n` HOC is not for your highest-level, it's for actual components.
2. You should use `withNamespaces` instead of `withI18n`.

[Directly from the docs](https://react.i18next.com/components/withi18n):

<img width="774" alt="screenshot 2018-11-07 at 19 32 12" src="https://user-images.githubusercontent.com/10575782/48176526-40e53d00-e2c5-11e8-99c5-a697a1464a12.png">

And further content from the docs:

> The withNamespaces hoc is responsible for passing the t function to your component. It enables all the translation functionality provided by i18next. Further, it asserts your component gets re-rendered on language change or changes to the translation catalog itself (loaded translations).

In short, if you want to ensure rerender on language change, use `withNamespaces`.

If you have any other questions, please post them on StackOverflow.

**Comment by wheelo** [comment]
yeah, I forget the export

**Comment by wheelo** [comment]
myriad of thanks
